### PR TITLE
google-cloud-bigtable-tool: init at 0.12.0

### PR DESCRIPTION
**Pull request by totoroot**
## Description of changes

- Adds a new package `google-cloud-bigtable-tool` at latest version 0.12.0

  The main program of the package is `cbt`.
  
  `cbt` is the Google Cloud Bigtable Tool. A CLI utility to interact with Google Cloud Bigtable.
  The cbt CLI is a command-line interface for performing several different operations on Cloud Bigtable.
  An overview of it's usage can be found in the [Google Cloud docs](https://cloud.google.com/bigtable/docs/cbt-overview).

- Renames package `cloud-sql-proxy` to `google-cloud-sql-proxy` as it cannot be used with other cloud providers (also to improve consistency)
  I added a note to the release notes of 23.11 as renaming a package might be considered breaking.
- Adds meta attributes to package `google-cloud-sql-proxy`

Ping maintainer @nicknovitski @totoroot 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by totoroot** [comment]
Result of `nixpkgs-review pr 261259` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>google-cloud-bigtable-tool</li>
    <li>google-cloud-sql-proxy</li>
  </ul>
</details>

**Comment by totoroot** [comment]
> Result of `nixpkgs-review pr 261259` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
> 1 package blacklisted:
> 
>     * nixos-install-tools
> 
> 
> 2 packages built:

Not sure what this means...
Perhaps that is normal for changes to the release notes.

**Comment by Janik-Haag** [comment]
Result of `nixpkgs-review pr 261259` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>google-cloud-bigtable-tool</li>
    <li>google-cloud-sql-proxy</li>
  </ul>
</details>

File: pkgs/tools/misc/google-cloud-bigtable-tool/default.nix [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+  src = fetchFromGitHub {
+    owner = "googleapis";
+    repo = "cloud-bigtable-cbt-cli";
+    rev = "v.${version}";
+    hash = "sha256-N5nbWMj7kLIdRiwBUWFz4Rat88Wx01i3hceMxAvSjaA=";
+  };
+
+  subPackages = [ "." ];
+
```
Comment (Janik-Haag) [inline pkgs/tools/misc/google-cloud-bigtable-tool/default.nix:18]:
```suggestion
```
You don't need this, the package gets build either ways.

**Comment by totoroot** [comment]
Removed the platforms meta attribute from both package definition, as I was informed that it is being automatically being set for all goModules and `platforms = platforms.all` was wrong anyways.

**Comment by Janik-Haag** [comment]
Result of `nixpkgs-review pr 261259` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>google-cloud-bigtable-tool</li>
    <li>google-cloud-sql-proxy</li>
  </ul>
</details>

**Review by Janik-Haag — APPROVED** [review]
LGTM

**Comment by 06kellyjac** [comment]
Hello, this one just bit me updating a project.

For future reference renames should be added to the aliases.nix file, ideally with an alias for a while, and then a throw that gives users a real warning not just an undefined variable.
thanks

**Comment by totoroot** [comment]
> Hello, this one just bit me updating a project.
> 
> For future reference renames should be added to the aliases.nix file, ideally with an alias for a while, and then a throw that gives users a real warning not just an undefined variable. thanks

@06kellyjac Sorry for that. Can do so going forward.
I thought mentioning it in the release notes would be sufficient.

**Comment by 06kellyjac** [comment]
Yeah it's not a big deal at all, just a minor inconvenience, I just thought I'd let you know as others might run into this.

**Comment by Janik-Haag** [comment]
Yeah, thanks I missed that one. I'll be more careful in the future.

**Pull request merged**